File: src/main/java/com/epri/dss/general/impl/CableDataObjImpl.java
package com.epri.dss.general.impl;

import java.io.PrintStream;

import com.epri.dss.common.DSSClass;
import com.epri.dss.general.CableDataObj;

public class CableDataObjImpl extends ConductorDataObjImpl implements CableDataObj {

	private double epsR;
	// next 3 use parent radiusUnits
	private double insLayer;
	private double diaIns;
	private double diaCable;

	public CableDataObjImpl(DSSClass parClass, String cableDataName) {
		super(parClass, cableDataName);

		setName(cableDataName.toLowerCase());
		objType = parClass.getDSSClassType();

		epsR     =  2.3;
		insLayer = -1.0;
		diaIns   = -1.0;
		diaCable = -1.0;
	}

	@Override
	public void dumpProperties(PrintStream f, boolean complete) {
		super.dumpProperties(f, complete);

		for (int i = 0; i < getParentClass().getNumProperties(); i++) {
			f.print("~ " + getParentClass().getPropertyName()[i] + "=");
			switch (i) {
			case 0:
				f.println(String.format("%.3g", epsR));
				break;
			case 1:
				f.println(String.format("%.6g", insLayer));
				break;
			case 2:
				f.println(String.format("%.6g", diaIns));
				break;
			case 3:
				f.println(String.format("%.6g", diaCable));
				break;
			}
		}
	}

	@Override
	public void initPropertyValues(int arrayOffset) {
		setPropertyValue(arrayOffset + 0, "2.3");  // TODO Check zero based indexing
		setPropertyValue(arrayOffset + 1, "-1");
		setPropertyValue(arrayOffset + 2, "-1");
		setPropertyValue(arrayOffset + 3, "-1");
		super.initPropertyValues(arrayOffset + 4);
	}

	public double getEpsR() {
		return epsR;
	}

	public double getInsLayer() {
		return insLayer;
	}

	public double getDiaIns() {
		return diaIns;
	}

	public double getDiaCable() {
		return diaCable;
	}

	// FIXME Private members in OpenDSS.

	public void setEpsR(double epsr) {
		epsR = epsr;
	}

	public void setInsLayer(double inslayer) {
		insLayer = inslayer;
	}

	public void setDiaIns(double diains) {
		diaIns = diains;
	}

	public void setDiaCable(double diacable) {
		diaCable = diacable;
	}

}

File: pydss/general/CableDataObjImpl.py
from dss.general.impl.ConductorDataObjImpl import (ConductorDataObjImpl,)
from dss.general.CableDataObj import (CableDataObj,)


class CableDataObjImpl(ConductorDataObjImpl, CableDataObj):
    epsR = None
    # next 3 use parent radiusUnits
    insLayer = None
    diaIns = None
    diaCable = None

    def __init__(self, parClass, cableDataName):
        super(CableDataObjImpl, self)(parClass, cableDataName)
        self.setName(cableDataName.toLowerCase())
        self.objType = parClass.getDSSClassType()
        self.epsR = 2.3
        self.insLayer = -1.0
        self.diaIns = -1.0
        self.diaCable = -1.0

    def dumpProperties(self, f, complete):
        super(CableDataObjImpl, self).dumpProperties(f, complete)
        _0 = True
        i = 0
        while True:
            if _0 is True:
                _0 = False
            else:
                i += 1
            if not (i < self.getParentClass().getNumProperties()):
                break
            f.print_('~ ' + self.getParentClass().getPropertyName()[i] + '=')
            _1 = i
            _2 = False
            while True:
                if _1 == 0:
                    _2 = True
                    print String.format.format('%.3g', self.epsR)
                    break
                if (_2 is True) or (_1 == 1):
                    _2 = True
                    print String.format.format('%.6g', self.insLayer)
                    break
                if (_2 is True) or (_1 == 2):
                    _2 = True
                    print String.format.format('%.6g', self.diaIns)
                    break
                if (_2 is True) or (_1 == 3):
                    _2 = True
                    print String.format.format('%.6g', self.diaCable)
                    break
                break

    def initPropertyValues(self, arrayOffset):
        self.setPropertyValue(arrayOffset + 0, '2.3')
        # TODO Check zero based indexing
        self.setPropertyValue(arrayOffset + 1, '-1')
        self.setPropertyValue(arrayOffset + 2, '-1')
        self.setPropertyValue(arrayOffset + 3, '-1')
        super(CableDataObjImpl, self).initPropertyValues(arrayOffset + 4)

    def getEpsR(self):
        return self.epsR

    def getInsLayer(self):
        return self.insLayer

    def getDiaIns(self):
        return self.diaIns

    def getDiaCable(self):
        # FIXME Private members in OpenDSS.
        return self.diaCable

    def setEpsR(self, epsr):
        self.epsR = epsr

    def setInsLayer(self, inslayer):
        self.insLayer = inslayer

    def setDiaIns(self, diains):
        self.diaIns = diains

    def setDiaCable(self, diacable):
        self.diaCable = diacable
